core/src/agent/keys.rs

```diff
@@ -1,9 +1,78 @@
 #[derive(Clone, Debug, PartialEq)]
 pub struct Key {}
 
+impl Key {
+
+    pub fn new() -> Key {
+        Key{}
+    }
+
+}
+
 #[derive(Clone, Debug, PartialEq)]
 pub struct Keys {
     pub_key: Key,
     priv_key: Key,
     node_id: String,
 }
+
+impl Keys {
+
+    pub fn new(pub_key: &Key, priv_key: &Key, node_id: &str) -> Keys {
```

Comment:
I think it would be easier to use a `S: Into<String>`.

Reply:
@sphinxc0re yar, it works, just seems like a lot of type juggling to support string arguments